lib/spreadsheet.ts

```diff
@@ -93,7 +93,13 @@ export function readSheet(
 		templateSheetName?: string;
 		clearSheet?: boolean;
 		overwriteRange?: boolean;
-	} = {},
+	} = {
```

Comment:
I don't know if there's a point in trying to decipher what Somo tried to convey almost a year ago. But we can try.

AFAICT, the core construct in action is
```typescript
parameterName: parameterType = defaultValue
```
(using placeholder names as an example)

However, since the parameter is a mapping of several parameters being unpacked on the go, things get weird. If the `parameterName` already includes default values then it's certainly odd to duplicate the same name-value pairs as `defaultValue`.

Maybe the `parameterType` could be defined separately as 
```typescript
parameterType = {
		startRow?: number;
		startColumn?: number;
		ss?: GoogleAppsScript.Spreadsheet.Spreadsheet;
		templateSheetName?: string;
		mode?: 'OVERWRITE' | 'APPENDROWS' | 'APPENDCOLUMNS';
		clearSheet?: boolean;
	}
```
and then
```typescript
export function writeToSheet(
	sheetName: string,
	writeArray: any[][],
	parameterName: parameterType = {
		startRow: 1,
		startColumn: 1,
		ss: SpreadsheetApp.getActiveSpreadsheet(),
		mode: 'OVERWRITE',
		clearSheet: false,
       },
) {
...
```
would be less weird.

Honestly, I don't think I have enough TS knowledge to provide recommendation here. Maybe we could ping some experts?

Reply:
> I'm happy to continue with what we have, unless you guys feel otherwise, then we can consult with TS experts on the team.

What we have looks reasonable to me @AlidaOdendaal. We can always discover improvements and incorporate those down the road. This doesn't look like a blocker anyway.